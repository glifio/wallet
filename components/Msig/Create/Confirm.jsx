import React, { useEffect, useRef } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { func, string } from 'prop-types'
import { useRouter } from 'next/router'
import ConfirmMessage from '../../../lib/confirm-message'
import {
  Card,
  Box,
  IconPending,
  StyledATag,
  Button,
  Text,
  Glyph,
  Title,
} from '../../Shared'
import { fetchAndSetMsigActor } from '../../../store/actions'

const NextOption = ({ text, onClick }) => {
  return (
    <Button
    title={text}
      display='flex'
      flexWrap='wrap'
      alignItems='center'
      mr={2}
      my={3}
      bg='core.transparent'
      borderColor='core.primary'
      color='core.primary'
      opacity='1'
      onClick={onClick}
      css={`
        &:hover {
          cursor: pointer;
        }
      `}
    />
  )
}

NextOption.propTypes = {
  text: string.isRequired,
  onClick: func.isRequired
}

const Confirm = () => {
  const dispatch = useDispatch()
  const { pending, confirmed } = useSelector(s => s.messages)
  const msgCid = useRef(pending[0]?.cid)
  const msigAddr = useSelector(s => s.msigActorAddress)
  const router = useRouter()

  useEffect(() => {
    if (confirmed.some(m => m.cid === msgCid.current)) {
      dispatch(fetchAndSetMsigActor(msgCid.current))
    }
  }, [confirmed, dispatch])

  return (
    <Box
      display='flex'
      justifyContent='center'
      width='100%'
      minHeight='100vh'
      p={3}
    >
      {msigAddr ? (
        <Box p={5}>
          <Title>Your multisig has been created.</Title>
            <Card maxWidth={13} width='100%' my={3} bg='background.screen' boxShadow={2} border={0}>
            <Glyph acronym='Ms' />
            <Text my={0} mt={3} color='core.darkgray'>Your Address </Text>
            <Text mt={2}>{msigAddr}</Text>
            </Card>
          <Text>What would you like to do?</Text>
          <Box display='flex' justifyContent='space-between'>
            <NextOption
              text='Go to Multisig home'
              onClick={() => {
                const searchParams = new URLSearchParams(router.query)
                router.push(`/vault/home?${searchParams.toString()}`)
              }}
            />
            <NextOption
              text='Print multisig details'
              onClick={() => {
                const searchParams = new URLSearchParams({
                  ...router.query,
                  address: msigAddr
                })
                router.push(`/vault/print?${searchParams.toString()}`)
              }}
            />
          </Box>
        </Box>
      ) : (
        <>
          <ConfirmMessage />
          {pending.length > 0 && (
            <Box display='flex' justifyContent='center' flexDirection='column'>
              <Box display='flex' justifyContent='center' alignItems='center'>
                <IconPending />
                <Text ml={2}>
                  We&apos;re waiting for your transaction to confirm.
                </Text>
              </Box>
              <Box display='flex' justifyContent='center' alignItems='center'>
                <Text mr={2}>With CID: </Text>
                <StyledATag
                  href={`https://filfox.info/en/message/${msgCid.current}`}
                >
                  {msgCid.current}
                </StyledATag>
                <br />
              </Box>
              <Text>
                This screen will automatically show you your new Multisig wallet
                address once the transaction confirms.
              </Text>
            </Box>
          )}
        </>
      )}
    </Box>
  )
}

export default Confirm
