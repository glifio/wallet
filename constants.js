/* WALLET TYPES */
export const LEDGER = 'LEDGER'
export const HD_WALLET = 'HD_WALLET'
export const SINGLE_KEY = 'SINGLE_KEY'

export const IMPORT_MNEMONIC = 'IMPORT_MNEMONIC'
export const CREATE_MNEMONIC = 'CREATE_MNEMONIC'
export const IMPORT_SINGLE_KEY = 'IMPORT_SINGLE_KEY'

/* API ENDPOINTS */
export const FILSCAN = 'https://forceup.io:8700/v0/filscan'
