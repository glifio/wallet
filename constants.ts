import { FilecoinNumber } from '@glif/filecoin-number'

/* WALLET TYPES */
export const LEDGER = 'LEDGER'
export const HD_WALLET = 'HD_WALLET'
export const SINGLE_KEY = 'SINGLE_KEY'

export const IMPORT_MNEMONIC = 'IMPORT_MNEMONIC'
export const CREATE_MNEMONIC = 'CREATE_MNEMONIC'
export const IMPORT_SINGLE_KEY = 'IMPORT_SINGLE_KEY'

/* API ENDPOINTS */
export const FILSCAN = 'https://api.filscan.io:8700/v0/filscan'
export const FILSCAN_JSONRPC = 'https://api.filscan.io:8700/rpc/v1'
export const FILSCOUT = 'https://filscoutv3api.ipfsunion.cn'
export const FILFOX = 'https://filfox.info/api'

/* NETWORK VARS */
export const MAINNET = 'f'
export const TESTNET = 't'

export const MAINNET_PATH_CODE = 461
export const TESTNET_PATH_CODE = 1

export const EXEC_ACTOR = 'f01'

/* TX METHOD TYPES */
export const SEND = 'SEND'
export const PROPOSE = 'PROPOSE'
export const EXEC = 'EXEC'
/* FILECOIN APP VERSION MIN */
export const LEDGER_VERSION_MAJOR = 0
export const LEDGER_VERSION_MINOR = 18
export const LEDGER_VERSION_PATCH = 2

/* GAS CONSTANTS */
export const emptyGasInfo = {
  estimatedTransactionFee: new FilecoinNumber('0', 'attofil'),
  gasPremium: new FilecoinNumber('0', 'attofil'),
  gasFeeCap: new FilecoinNumber('0', 'attofil'),
  gasLimit: new FilecoinNumber('0', 'attofil')
}

/* MSIG PL SIGNERS */
export const PL_SIGNERS = new Set([
  'f020484',
  'f022215',
  'f023047',
  'f022408',
  'f0105',
  'f125nlfn4lwl7ldt4gwfas5furnwafwqgd2our6oa',
  'f16j5fivw6dq2h3uujq47joz36on24wfalofxmtvi',
  'f1yr7w7y5nia2zmzkw6yvxygj2pctwv3le2l3zccq',
  'f1figgwd5npr2jyw67u75qhjoyv3xf3jq5vwmmq5i',
  'f1ovvm6oilbdsvbw27jhil3pcywrjuwiv5uzagq6i',
  'f010361',
  't020484',
  't022215',
  't023047',
  't022408',
  't0105',
  't125nlfn4lwl7ldt4gwfas5furnwafwqgd2our6oa',
  't16j5fivw6dq2h3uujq47joz36on24wfalofxmtvi',
  't1yr7w7y5nia2zmzkw6yvxygj2pctwv3le2l3zccq',
  't1figgwd5npr2jyw67u75qhjoyv3xf3jq5vwmmq5i',
  't1ovvm6oilbdsvbw27jhil3pcywrjuwiv5uzagq6i'
])

/* STYLE CONSTANTS */
export const SCREEN_MAX_WIDTH = 1440

/* PAGES */
/* eslint-disable no-unused-vars */
export enum PAGE {
  WALLET_HOME = '/home',
  WALLET_SEND = '/send',
  SPEED_UP = '/speed-up',
  MSIG_HOME = '/vault/home',
  MSIG_HISTORY = '/vault/history',
  MSIG_ADMIN = '/vault/admin',
  MSIG_WITHDRAW = '/vault/withdraw',
  MSIG_CHANGE_SIGNER = '/vault/change-signer',
  MSIG_REMOVE_SIGNER = '/vault/remove-signer',
  MSIG_ADD_SIGNER = '/vault/add-signer',
  MSIG_CREATE_CONFIRM = '/vault/create/confirm',
  MSIG_CREATE = '/vault/create'
}

export enum MSIG_METHOD {
  CONSTRUCTOR = 1,
  PROPOSE,
  APPROVE,
  CANCEL,
  ADD_SIGNER,
  REMOVE_SIGNER,
  SWAP_SIGNER,
  CHANGE_NUM_APPROVALS_THRESHOLD,
  LOCK_BALANCE
}
/* eslint-enable */

// todo #responsiveDesign: decide how to do responsive design
export const RESPONSIVE_BREAKPOINT = 1024
