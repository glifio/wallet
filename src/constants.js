/* WALLET TYPES */
export const LEDGER = 'LEDGER'
