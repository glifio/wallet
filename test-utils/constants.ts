export const WALLET_ADDRESS = 't1z225tguggx4onbauimqvxzutopzdr2m4s6z6wgi'
