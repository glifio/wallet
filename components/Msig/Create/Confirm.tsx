import React, { SyntheticEvent, useState } from 'react'
import { func, string } from 'prop-types'
import { useRouter } from 'next/router'
import {
  Card,
  Box,
  IconPending,
  StyledATag,
  Button,
  Text,
  Glyph,
  Title
} from '@glif/react-components'

import { useMsig } from '../../../MsigProvider'
import { PAGE } from '../../../constants'
import { navigate } from '../../../utils/urlParams'

const NextOption = ({
  text,
  onClick
}: {
  text: string
  onClick: (_: SyntheticEvent) => void
}) => {
  return (
    <Button
      title={text}
      display='flex'
      flexWrap='wrap'
      alignItems='center'
      mr={2}
      my={3}
      bg='core.transparent'
      borderColor='core.primary'
      color='core.primary'
      opacity='1'
      onClick={onClick}
      css={`
        &:hover {
          cursor: pointer;
        }
      `}
    />
  )
}

NextOption.propTypes = {
  text: string.isRequired,
  onClick: func.isRequired
}

const Confirm = () => {
  const [msigError, setMsigError] = useState('')
  const { setMsigActor, Address } = useMsig()
  const router = useRouter()
  const msgCid = ''

  if (msigError) {
    return (
      <Box
        display='flex'
        justifyContent='center'
        width='100%'
        minHeight='100vh'
        p={3}
      >
        <Box display='flex' justifyContent='center' flexDirection='column'>
          <Box display='flex' justifyContent='center' alignItems='center'>
            <Title ml={2}>
              There was an error when creating your multisig.
            </Title>
          </Box>
          <Box display='flex' justifyContent='center' alignItems='center'>
            <Text mr={2}>With CID: </Text>
            <StyledATag href={`https://filfox.info/en/message/${msgCid}`}>
              {msgCid}
            </StyledATag>
            <br />
          </Box>
        </Box>
      </Box>
    )
  }

  return (
    <Box
      display='flex'
      justifyContent='center'
      width='100%'
      minHeight='100vh'
      p={3}
    >
      {Address ? (
        <Box
          p={5}
          display='flex'
          flexDirection='column'
          alignItems='center'
          justifyContent='center'
        >
          <Title>Your multisig has been created.</Title>
          <Card
            maxWidth={13}
            width='100%'
            my={3}
            bg='background.screen'
            boxShadow={2}
            border={0}
          >
            <Glyph acronym='Ms' />
            <Text my={0} mt={3} color='core.darkgray'>
              Your Address{' '}
            </Text>
            <Text mt={2}>{Address}</Text>
          </Card>
          <NextOption
            text='Go to Multisig home'
            onClick={() => {
              navigate(router, { pageUrl: PAGE.MSIG_HOME })
            }}
          />
        </Box>
      ) : (
        <>
          {pending.length > 0 && (
            <Box display='flex' justifyContent='center' flexDirection='column'>
              <Box display='flex' justifyContent='center' alignItems='center'>
                <IconPending />
                <Text ml={2}>
                  We&apos;re waiting for your transaction to confirm.
                </Text>
              </Box>
              <Box display='flex' justifyContent='center' alignItems='center'>
                <Text mr={2}>With CID: </Text>
                <StyledATag href={`https://filfox.info/en/message/${msgCid}`}>
                  {msgCid}
                </StyledATag>
                <br />
              </Box>
              <Text>
                This screen will automatically show you your new Multisig wallet
                address once the transaction confirms.
              </Text>
            </Box>
          )}
        </>
      )}
    </Box>
  )
}

export default Confirm
