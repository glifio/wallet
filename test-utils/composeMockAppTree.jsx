/* eslint-disable react/prop-types */
import React, { useReducer } from 'react'
import cloneDeep from 'lodash.clonedeep'

import WalletProvider from '@openworklabs/filecoin-wallet-provider'
import { FilecoinNumber } from '@openworklabs/filecoin-number'
import { Provider } from 'react-redux'

import { NetworkCheck } from '../lib/check-network'
import BalancePoller from '../lib/update-balance'
import { ConverterContext } from '../lib/Converter'
import { theme, ThemeProvider } from '../components/Shared'
import { initializeStore } from './index'
import { initialState } from '../store/states'
import walletProviderReducer, {
  initialState as walletProviderInitialState,
  setWalletType,
  setError,
  resetLedgerState,
  resetState
} from '../WalletProvider/state'
import { WalletProviderContext } from '../WalletProvider'
import createPath from '../utils/createPath'
import { IMPORT_MNEMONIC } from '../constants'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

const mockRouter = jest.fn(() => {})
useRouter.mockImplementation(() => ({
  query: 'network=t',
  push: mockRouter,
  replace: mockRouter
}))

jest.mock('@openworklabs/filecoin-wallet-provider')
const mockGetAccounts = jest
  .fn()
  .mockImplementation(() =>
    Promise.resolve(['t1mbk7q6gm4rjlndfqw6f2vkfgqotres3fgicb2uq'])
  )

const mockGetBalance = jest
  .fn()
  .mockImplementation(() => Promise.resolve(new FilecoinNumber('1', 'fil')))

WalletProvider.mockImplementation(() => {
  return {
    wallet: {
      getAccounts: mockGetAccounts,
      sign: jest.fn().mockImplementation(() => 'xxxyyyyzzzz')
    },
    getBalance: mockGetBalance,
    getNonce: jest.fn().mockImplementation(() => 0),
    estimateGas: jest
      .fn()
      .mockImplementation(() => new FilecoinNumber('122', 'attofil')),
    sendMessage: jest.fn().mockImplementation(() => 'QmZCid!')
  }
})

const mockWalletProviderInstance = new WalletProvider()

const presets = {
  preOnboard: cloneDeep(initialState),
  postOnboard: cloneDeep({
    ...initialState,
    wallets: [
      {
        address: 't1z225tguggx4onbauimqvxzutopzdr2m4s6z6wgi',
        balance: new FilecoinNumber('1', 'fil'),
        path: createPath(1, 0)
      }
    ],
    selectedWalletIdx: 0
  })
}

const composeWalletProviderState = (initialWalletProviderState, preset) => {
  switch (preset) {
    case 'postOnboard': {
      return Object.freeze({
        ...initialWalletProviderState,
        walletType: IMPORT_MNEMONIC,
        walletProvider: mockWalletProviderInstance
      })
    }
    default:
      return initialWalletProviderState
  }
}

/**
 * VALID OPTIONS:
 * mockConverterInstance (an object that stubs the converter functionality)
 * walletProviderReducer (a reducer to stub the state handling of the wallet provider)
 * walletProviderInitialState(a reducer to stub the state of the walletProvider context)
 * pathname
 * query (usually { network: 'f' } or { network: 't' })
 */

export default (statePreset = 'preOnboard', options = {}) => {
  const state = options.state || presets[statePreset]
  const store = initializeStore(state)

  const mockConverterInstance = options.mockConverterInstance || {
    cacheConversionRate: jest.fn(),
    toFIL: jest.fn().mockImplementation(amount => {
      return new FilecoinNumber(amount, 'fil').dividedBy(5)
    }),
    fromFIL: jest.fn().mockImplementation(amount => {
      return new FilecoinNumber(amount, 'fil').multipliedBy(5)
    })
  }

  const pathname = options.pathname || '/wallet'
  const query = options.query || { network: 't' }

  const Tree = ({ children }) => {
    const [initialWalletProviderState, walletProviderDispatch] = useReducer(
      options.reducer || walletProviderReducer,
      options.walletProviderInitialState || walletProviderInitialState
    )

    const walletProviderState = composeWalletProviderState(
      initialWalletProviderState,
      statePreset
    )
    return (
      <Provider store={store}>
        <ConverterContext.Provider
          value={{
            converter: mockConverterInstance,
            converterError: options.converterError || null
          }}
        >
          <WalletProviderContext.Provider
            value={{
              state: walletProviderState,
              dispatch: walletProviderDispatch,
              fetchDefaultWallet: jest.fn().mockImplementation(),
              setWalletError: jest
                .fn()
                .mockImplementation(errorMessage =>
                  walletProviderDispatch(setError(errorMessage))
                ),
              setWalletType: jest
                .fn()
                .mockImplementation(walletType =>
                  walletProviderDispatch(setWalletType(walletType))
                ),
              setLedgerProvider: jest.fn(),
              connectLedger: jest.fn(),
              resetLedgerState: jest.fn().mockImplementation(() => {
                walletProviderDispatch(resetLedgerState())
              }),
              resetState: jest.fn().mockImplementation(() => {
                walletProviderDispatch(resetState())
              })
            }}
          >
            <NetworkCheck
              networkFromRdx={store.getState().network}
              pathname={pathname}
              query={query}
              switchNetwork={jest.fn()}
            />
            <BalancePoller />
            <ThemeProvider theme={theme}>{children}</ThemeProvider>
          </WalletProviderContext.Provider>
        </ConverterContext.Provider>
      </Provider>
    )
  }

  return { Tree, store, walletProvider: mockWalletProviderInstance }
}
