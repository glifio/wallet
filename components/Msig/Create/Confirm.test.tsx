import '@testing-library/jest-dom/extend-expect'
import { act, cleanup, render, screen } from '@testing-library/react'

import composeMockAppTree from '../../../test-utils/composeMockAppTree'
import { PAGE } from '../../../constants'

import Confirm from './Confirm'
import { flushPromises } from '../../../test-utils'

jest.mock('../../../MsigProvider')
const mockMessageConfirmation = jest.fn(async () => true)
jest
  .spyOn(require('@glif/filecoin-message-confirmer'), 'default')
  .mockImplementation(mockMessageConfirmation)

const mockStateGetMessageReceipt = jest.fn(async () => ({
  ExitCode: 0,
  Return: 'gkQAyogBVQJnC6HX7N9krE3QzLET9tDO1XuUqw==',
  GasUsed: 8555837
}))
jest
  .spyOn(require('@glif/filecoin-rpc-client'), 'default')
  .mockImplementation(() => {
    return {
      request: mockStateGetMessageReceipt
    }
  })

jest.spyOn(require('next/router'), 'useRouter').mockImplementation(() => {
  return {
    query: { network: 't' },
    pathname: PAGE.MSIG_CREATE_CONFIRM,
    push: jest.fn()
  }
})

describe('confirmation of newly created multisig', () => {
  afterEach(cleanup)
  beforeEach(() => {
    jest.clearAllMocks()
  })

  test('it renders message pending UI while the transaction is pending', () => {
    const { Tree } = composeMockAppTree('pendingMsigCreate')
    let container, unmount
    act(() => {
      const rendered = render(
        <Tree>
          <Confirm />
        </Tree>
      )

      container = rendered.container
      unmount = rendered.unmount
    })

    expect(
      screen.getByText(/waiting for your transaction to confirm./)
    ).toBeInTheDocument()
    expect(
      screen.getByText(
        /This screen will automatically show you your new Multisig wallet address once the transaction confirms./
      )
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
    unmount()
  })

  test('it attempts to confirm the pending msig create message', async () => {
    const { Tree } = composeMockAppTree('pendingMsigCreate')
    let unmount
    act(() => {
      const rendered = render(
        <Tree>
          <Confirm />
        </Tree>
      )
      unmount = rendered.unmount
    })

    expect(mockMessageConfirmation).toHaveBeenCalled()
    unmount()
  })

  test('it displays the multisig address after the message is confirmed', async () => {
    const { Tree } = composeMockAppTree('pendingMsigCreate')
    let container, unmount
    await act(async () => {
      const rendered = render(
        <Tree>
          <Confirm />
        </Tree>
      )
      container = rendered.container
      unmount = rendered.unmount
    })
    await flushPromises()

    expect(mockMessageConfirmation).toHaveBeenCalled()
    expect(screen.getByText(/Your multisig has been created./))
    expect(container.firstChild).toMatchSnapshot()
    unmount()
  })
})
