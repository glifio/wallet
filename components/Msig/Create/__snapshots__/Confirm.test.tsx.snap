// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`confirmation of newly created multisig it displays the multisig address after the message is confirmed 1`] = `
.c0 {
  min-width: 0;
  box-sizing: border-box;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  min-height: 100vh;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  min-width: 0;
  box-sizing: border-box;
  padding: 32px;
}

.c3 {
  min-width: 0;
  box-sizing: border-box;
  border: 0;
  border-radius: 4px;
  border-width: 1px;
  padding: 16px;
  margin-top: 16px;
  margin-bottom: 16px;
  background-color: #EFF3FD;
  display: inline-block;
  width: 100%;
  overflow: hidden;
  max-width: 560px;
  box-shadow: rgba(0,0,0,0.10) 0px 0.7px 2.2px -8px,rgba(0,0,0,0.04) 0px 1.7px 2.4px,rgba(0,0,0,0.106) 0px 3.1px 8.1px,rgba(0,0,0,0.04) 0px 5.6px 12.1px,rgba(0,0,0,0.045) 0px 4.4px 4.8px,rgba(0,0,0,0.05) 0px 15px 41px;
}

.c4 {
  min-width: 0;
  box-sizing: border-box;
  border-width: 0.1875rem;
  border-radius: 8px;
  border-style: solid;
  color: #000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 48px;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c9 {
  min-width: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  font-family: "RT-Alias-Medium","system-ui","Segoe UI","Roboto",Helvetica;
  font-weight: 700;
  font-size: 1.5rem;
}

.c10 {
  cursor: pointer;
  border: 1px solid core.primary;
  background-color: #1AD08F;
  border-color: core.primary;
  color: core.primary;
  font-size: 1.125rem;
  -webkit-transition: 0.18s ease-in-out;
  transition: 0.18s ease-in-out;
  border-radius: 4px;
  margin-right: 8px;
  margin-top: 16px;
  margin-bottom: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 48px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-color: #5E26FF;
  border-width: 1px;
  border-radius: 4px;
  background-color: transparent;
  color: #5E26FF;
  opacity: 1;
}

.c10:hover {
  opacity: 0.8;
}

.c2 {
  font-size: 1.5rem;
  font-weight: 400;
  line-height: 1.2;
  font-family: RT-Alias-Grotesk;
  margin: 0;
}

.c8 {
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
}

.c6 {
  color: #666666;
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
  margin-top: 16px;
  margin-bottom: 0;
}

.c7 {
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
  margin-top: 8px;
}

.c11:hover {
  cursor: pointer;
}

<div
  class="c0"
  display="flex"
  width="100%"
>
  <div
    class="c1"
  >
    <h2
      class="c2"
      font-family="RT-Alias-Grotesk"
      font-size="4"
      font-weight="400"
    >
      Your multisig has been created.
    </h2>
    <div
      class="c3"
      display="inline-block"
      overflow="hidden"
      width="100%"
    >
      <div
        class="c4"
        color="#000"
        display="flex"
        size="6"
      >
        <h3
          class="c5"
        >
          Ms
        </h3>
      </div>
      <p
        class="c6"
        color="core.darkgray"
        font-family="RT-Alias-Grotesk"
        font-size="2"
        font-weight="400"
      >
        Your Address
         
      </p>
      <p
        class="c7"
        font-family="RT-Alias-Grotesk"
        font-size="2"
        font-weight="400"
      >
        f2m4f2dv7m35skytoqzsyrh5wqz3kxxfflxsha5za
      </p>
    </div>
    <p
      class="c8"
      font-family="RT-Alias-Grotesk"
      font-size="2"
      font-weight="400"
    >
      What would you like to do?
    </p>
    <div
      class="c9"
      display="flex"
    >
      <button
        class="c10 c11 Confirm___StyledButton-sc-1z08vvm-0 c11"
        color="core.primary"
        display="flex"
        font-size="3"
        height="6"
        opacity="1"
        type="button"
      >
        Go to Multisig home
      </button>
    </div>
  </div>
</div>
`;

exports[`confirmation of newly created multisig it renders message pending UI while the transaction is pending 1`] = `
.c0 {
  min-width: 0;
  box-sizing: border-box;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  min-height: 100vh;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  min-width: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  min-width: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
  margin-left: 8px;
}

.c6 {
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
  margin-right: 8px;
}

.c8 {
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
}

.c3 {
  width: 24px;
  height: 24px;
}

.c4 {
  -webkit-animation: WNDHq 4s infinite;
  animation: WNDHq 4s infinite;
}

.c7 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: 0.18s ease-in-out;
  transition: 0.18s ease-in-out;
  border-bottom: 2px solid #5E26FF00;
  color: #5E26FF;
  font-size: 1.125rem;
}

.c7:hover {
  border-bottom: 2px solid #5E26FF;
}

<div
  class="c0"
  display="flex"
  width="100%"
>
  <div
    class="c1"
    display="flex"
  >
    <div
      class="c2"
      display="flex"
    >
      <svg
        class="c3 c4"
        fill="none"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#FDB221"
          height="24"
          rx="12"
          width="24"
        />
        <mask
          fill="white"
          id="path-2-inside-1"
        >
          <path
            clip-rule="evenodd"
            d="M11.4006 11.9985L7.08534 6.60939C6.88876 6.36388 7.06355 6 7.37806 6H16.6265C16.9411 6 17.1159 6.36388 16.9193 6.60939L12.6017 12.0015L16.9169 17.3906C17.1135 17.6361 16.9387 18 16.6242 18H7.3757C7.06119 18 6.8864 17.6361 7.08298 17.3906L11.4006 11.9985Z"
            fill-rule="evenodd"
          />
        </mask>
        <path
          clip-rule="evenodd"
          d="M11.4006 11.9985L7.08534 6.60939C6.88876 6.36388 7.06355 6 7.37806 6H16.6265C16.9411 6 17.1159 6.36388 16.9193 6.60939L12.6017 12.0015L16.9169 17.3906C17.1135 17.6361 16.9387 18 16.6242 18H7.3757C7.06119 18 6.8864 17.6361 7.08298 17.3906L11.4006 11.9985Z"
          fill="#FDB221"
          fill-rule="evenodd"
        />
        <path
          d="M11.4006 11.9985L12.5715 12.9361L13.3222 11.9985L12.5715 11.061L11.4006 11.9985ZM7.08534 6.60939L5.91446 7.54695L5.91446 7.54695L7.08534 6.60939ZM16.9193 6.60939L15.7484 5.67183L15.7484 5.67183L16.9193 6.60939ZM12.6017 12.0015L11.4308 11.0639L10.6801 12.0015L11.4308 12.939L12.6017 12.0015ZM16.9169 17.3906L15.746 18.3282L15.746 18.3282L16.9169 17.3906ZM7.08298 17.3906L8.25387 18.3282L8.25387 18.3282L7.08298 17.3906ZM12.5715 11.061L8.25623 5.67183L5.91446 7.54695L10.2297 12.9361L12.5715 11.061ZM8.25623 5.67183C8.84599 6.40835 8.32162 7.5 7.37806 7.5V4.5C5.80547 4.5 4.93153 6.31941 5.91446 7.54695L8.25623 5.67183ZM7.37806 7.5H16.6265V4.5H7.37806V7.5ZM16.6265 7.5C15.683 7.5 15.1586 6.40836 15.7484 5.67183L18.0902 7.54695C19.0731 6.3194 18.1991 4.5 16.6265 4.5V7.5ZM15.7484 5.67183L11.4308 11.0639L13.7726 12.939L18.0902 7.54695L15.7484 5.67183ZM11.4308 12.939L15.746 18.3282L18.0878 16.453L13.7726 11.0639L11.4308 12.939ZM15.746 18.3282C15.1563 17.5916 15.6806 16.5 16.6242 16.5V19.5C18.1968 19.5 19.0707 17.6806 18.0878 16.453L15.746 18.3282ZM16.6242 16.5H7.3757V19.5H16.6242V16.5ZM7.3757 16.5C8.31925 16.5 8.84363 17.5916 8.25387 18.3282L5.9121 16.453C4.92916 17.6806 5.80312 19.5 7.3757 19.5V16.5ZM8.25387 18.3282L12.5715 12.9361L10.2297 11.061L5.9121 16.453L8.25387 18.3282Z"
          fill="#444444"
          mask="url(#path-2-inside-1)"
        />
      </svg>
      <p
        class="c5"
        font-family="RT-Alias-Grotesk"
        font-size="2"
        font-weight="400"
      >
        We're waiting for your transaction to confirm.
      </p>
    </div>
    <div
      class="c2"
      display="flex"
    >
      <p
        class="c6"
        font-family="RT-Alias-Grotesk"
        font-size="2"
        font-weight="400"
      >
        With CID: 
      </p>
      <a
        class="c7"
        color="core.primary"
        font-size="2"
        href="https://filfox.info/en/message/bafy2bzaced2godar6dht6ag6omfa53jr6tykygtzskzodqo2k4ngvyou4yct2"
        target="_blank"
      >
        bafy2bzaced2godar6dht6ag6omfa53jr6tykygtzskzodqo2k4ngvyou4yct2
      </a>
      <br />
    </div>
    <p
      class="c8"
      font-family="RT-Alias-Grotesk"
      font-size="2"
      font-weight="400"
    >
      This screen will automatically show you your new Multisig wallet address once the transaction confirms.
    </p>
  </div>
</div>
`;
